foundation/device/device-id/src/main/kotlin/com/pingidentity/device/id/DeviceIdentifier.kt

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright (c) 2025 Ping Identity Corporation. All rights reserved.
+ *
+ * This software may be modified and distributed under the terms
+ * of the MIT license. See the LICENSE file for details.
+ */
+
+package com.pingidentity.device.id
+
+/**
+ * Interface for obtaining a device-specific identifier.
+ *
+ * Implementations of this interface provide different strategies for generating
+ * or retrieving identifiers that can uniquely identify a device. These identifiers
+ * can be used for device recognition, fraud detection, or other security purposes.
+ *
+ * The SDK provides several implementations of this interface:
+ * - [DefaultDeviceIdentifier]: Uses the Android KeyStore to generate a persistent identifier
+ * - [AndroidIDDeviceIdentifier]: Uses the device's Android ID
+ *
+ * Note: When implementing this interface, consider privacy implications and
+ * ensure compliance with relevant privacy regulations.
+ */
+interface DeviceIdentifier {
```

Comment:
Compared to iOS, it seems the Android implementation is entirely synchronous, which may block the UI thread during key generation operations.